SIGSEGV with OCI chart reference using helm template --dependency-update

Summary:

When generating helm templates using `helm template --dependency-update`, and no dependency charts have been previously downloaded, helm produces a SIGSEGV. Non-OCI dependencies download without issue in the same `Chart.yaml` definition.

Example command:
```
helm template charts/foo --dependency-update --output-dir /tmp/tmp.helmlint.tHqjfFIP/charts/foo --values charts/foo/ci/default-values.yaml
```

Output of `helm version`:
```
version.BuildInfo{Version:"v3.10.1", GitCommit:"9f88ccb6aee40b9a0535fcc7efea6055e1ef72c9", GitTreeState:"clean", GoVersion:"go1.18.7"}
```

Output of `kubectl version`:
```
Client Version: v1.25.3
Kustomize Version: v4.5.7
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): n/a

stacktrace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x176d164]
goroutine 1 [running]:
helm.sh/helm/v3/pkg/registry.(*Client).Pull(0x0, {0xc002b0c4e6?, 0xc012afd2c8?}, {0x0, 0x0, 0x3?})
	helm.sh/helm/v3/pkg/registry/client.go:303 +0x424
helm.sh/helm/v3/pkg/getter.(*OCIGetter).get(0xc012afd2e8?, {0xc002b0c4e0, 0x56})
	helm.sh/helm/v3/pkg/getter/ocigetter.go:53 +0x274
helm.sh/helm/v3/pkg/getter.(*OCIGetter).Get(0xc0007ba630, {0xc002b0c4e0, 0x56}, {0xc001626680, 0x6, 0x4?})
	helm.sh/helm/v3/pkg/getter/ocigetter.go:36 +0x70
helm.sh/helm/v3/pkg/downloader.(*ChartDownloader).DownloadTo(0xc012afd710, {0xc002b0c480, 0x47}, {0xc002b0c4c8?, 0x1?}, {0xc001dba340, 0x38})
	helm.sh/helm/v3/pkg/downloader/chart_downloader.go:100 +0xea
helm.sh/helm/v3/pkg/downloader.(*Manager).downloadAll(0xc000482000, {0xc00261d600?, 0x2b, 0x2b})
	helm.sh/helm/v3/pkg/downloader/manager.go:356 +0xebb
helm.sh/helm/v3/pkg/downloader.(*Manager).Update(0xc000482000)
	helm.sh/helm/v3/pkg/downloader/manager.go:201 +0x109
main.runInstall({0xc000a61800?, 0x1, 0x8}, 0xc0002d9b80, 0xc000b1fae8?, {0x207bf60?, 0xc00000e018})
	helm.sh/helm/v3/cmd/helm/install.go:248 +0x65e
main.newTemplateCmd.func2(0xc000116500?, {0xc000a61800?, 0x8?, 0x8?})
	helm.sh/helm/v3/cmd/helm/template.go:82 +0x22a
github.com/spf13/cobra.(*Command).execute(0xc000116500, {0xc000a61780, 0x8, 0x8})
	github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000a1ec80)
	github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	helm.sh/helm/v3/cmd/helm/helm.go:83 +0x235

```

--- a/cmd/helm/install.go
+++ b/cmd/helm/install.go
@@ -252,6 +252,7 @@ func runInstall(args []string, client *action.Install, valueOpts *values.Options
 					RepositoryConfig: settings.RepositoryConfig,
 					RepositoryCache:  settings.RepositoryCache,
 					Debug:            settings.Debug,
+					RegistryClient:   client.GetRegistryClient(),
 				}
 				if err := man.Update(); err != nil {
 					return nil, err

--- a/pkg/action/install.go
+++ b/pkg/action/install.go
@@ -142,6 +142,11 @@ func (i *Install) SetRegistryClient(registryClient *registry.Client) {
 	i.ChartPathOptions.registryClient = registryClient
 }
 
+// GetRegistryClient get the registry client.
+func (i *Install) GetRegistryClient() *registry.Client {
+	return i.ChartPathOptions.registryClient
+}
+
 func (i *Install) installCRDs(crds []chart.CRD) error {
 	// We do these one file at a time in the order they were read.
 	totalItems := []*resource.Info{}

